### PR TITLE
Use GCARR_ac rate law function for full-chemistry reactions where the Arrhenius B parameter is zero

### LINKED ISSUE #1641: [BUG/ISSUE] Several reactions in fullchem.eqn are using GCARR_abc when they should use GCARR_ac
### What institution are you from?
Harvard University 

### Description of the problem
I noticed several of the chemistry updates added to fullchem.eqn after GEOS-Chem 13.0.0 use the `GCARR_abc` function when they should use `GCARR_ac` because the B term is zero. From fullchem.eqn:

```
// (3) To avoid useless CPU cycles, we have introduced new rate law functions
//     that skip computing Arrhenius terms (and other terms) that would
//     evaluate to 1.  The Arrhenius terms that are passed to the function
//     are in most cases now noted in the function name (e.g. GCARR_abc takes
//     Arrhenius A, B, C parameters but GCARR_ac only passes A and C
//     parameters because B=0 and the (300/T)*B would evaluate to 1).
//     This should be much more computationally efficient, as these functions
//     are called (sometimes multiple times) for each grid box where we
//     perform chemistry.
//        -- Bob Yantosca (25 Jan 2020)
```

The reactions that can be modified to use `GCARR_ac` can mainly be found in these lines:

https://github.com/geoschem/geos-chem/blob/205995a40535e8d43dffc3d436bf640880ec4823/KPP/fullchem/fullchem.eqn#L1386-L1483

### GEOS-Chem version
GEOS-Chem 14.1.0

### PR DESCRIPTION
**Pull request by yantosca**
### Name and Institution (Required)

Name: Bob Yantosca
Institution: Harvard + GCST

### Confirm you have reviewed the following documentation

- [x] [Contributing guidelines](https://geos-chem.readthedocs.io/en/stable/reference/CONTRIBUTING.html)

### Describe the update
This PR addresses the issue raised in #1641.  Recent chemistry updates had brought in several reactions that used the rate-law function `GCARR_abc`, even though the Arrhenius 'B` parameter is zero for these reactions.  This results in raising 0 to a power, which will always evaluate to 1, causing useless computational overhead for the same result.

We have replaced these calls to `GCARR_abc` with calls to `GCARR_ac`.  This will produce the same result but skip the useless computations.  This should result in faster execution.

Also, we now point users to `KPP/fullchem/CHANGELOG_fullchem.md` for a list of updates tot the fullchem mechanism.

### Expected changes
This is expected to be a zero-diff update.

### Reference(s)
N/A

### Related Github Issue(s)
Closes #1641

### CODE REVIEW

**Comment by yantosca** [comment]
All GEOS-Chem Classic integration tests passed:
```console
==============================================================================
GEOS-Chem Classic: Execution Test Results

GCClassic #a44299c GEOS-Chem submodule update: Restore Iy sink reactions
GEOS-Chem #e0426b12f Use GCARR_ac rate law function for rxns with Arrhenius B = 0
HEMCO     #a4f8517 Merge PR #189 (branch 'feature/add-lumped-furans') into dev/3.7.0

Using 24 OpenMP threads
Number of execution tests: 26

Submitted as SLURM job: 46650121
==============================================================================
 
Execution tests:
------------------------------------------------------------------------------
gc_05x0625_NA_47L_merra2_CH4........................Execute Simulation....PASS
gc_05x0625_NA_47L_merra2_fullchem...................Execute Simulation....PASS
gc_4x5_47L_merra2_fullchem..........................Execute Simulation....PASS
gc_4x5_47L_merra2_fullchem_TOMAS15..................Execute Simulation....PASS
gc_4x5_47L_merra2_fullchem_TOMAS40..................Execute Simulation....PASS
gc_4x5_merra2_aerosol...............................Execute Simulation....PASS
gc_4x5_merra2_carbon................................Execute Simulation....PASS
gc_4x5_merra2_CH4...................................Execute Simulation....PASS
gc_4x5_merra2_CO2...................................Execute Simulation....PASS
gc_4x5_merra2_fullchem..............................Execute Simulation....PASS
gc_4x5_merra2_fullchem_aciduptake...................Execute Simulation....PASS
gc_4x5_merra2_fullchem_APM..........................Execute Simulation....PASS
gc_4x5_merra2_fullchem_benchmark....................Execute Simulation....PASS
gc_4x5_merra2_fullchem_complexSOA...................Execute Simulation....PASS
gc_4x5_merra2_fullchem_complexSOA_SVPOA.............Execute Simulation....PASS
gc_4x5_merra2_fullchem_LuoWd........................Execute Simulation....PASS
gc_4x5_merra2_fullchem_marinePOA....................Execute Simulation....PASS
gc_4x5_merra2_fullchem_RRTMG........................Execute Simulation....PASS
gc_4x5_merra2_Hg....................................Execute Simulation....PASS
gc_4x5_merra2_metals................................Execute Simulation....PASS
gc_4x5_merra2_POPs_BaP..............................Execute Simulation....PASS
gc_4x5_merra2_tagCH4................................Execute Simulation....PASS
gc_4x5_merra2_tagCO.................................Execute Simulation....PASS
gc_4x5_merra2_tagO3.................................Execute Simulation....PASS
gc_4x5_merra2_TransportTracers......................Execute Simulation....PASS
gc_4x5_merra2_TransportTracers_LuoWd................Execute Simulation....PASS
 
Summary of test results:
------------------------------------------------------------------------------
Execution tests passed: 26
Execution tests failed: 0
Execution tests not yet completed: 0

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%  All execution tests passed!  %%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
```

**Comment by yantosca** [comment]
All GCHP integration tests passed (except tagO3, which will be fixed later)
```console
==============================================================================
GCHP: Execution Test Results

GCClassic #0b6d0e2 GEOS-Chem submodule update: Restore Iy sink reactions
GEOS-Chem #e0426b12f Use GCARR_ac rate law function for rxns with Arrhenius B = 0
HEMCO     #a4f8517 Merge PR #189 (branch 'feature/add-lumped-furans') into dev/3.7.0

Number of execution tests: 5

Submitted as SLURM job: 46650313
==============================================================================
 
Execution tests:
------------------------------------------------------------------------------
gchp_merra2_fullchem................................Execute Simulation....PASS
gchp_merra2_fullchem_benchmark......................Execute Simulation....PASS
gchp_merra2_fullchem_RRTMG..........................Execute Simulation....PASS
gchp_merra2_tagO3...................................Execute Simulation....FAIL
gchp_merra2_TransportTracers........................Execute Simulation....PASS
 
Summary of test results:
------------------------------------------------------------------------------
Execution tests passed: 4
Execution tests failed: 1
Execution tests not yet completed: 0
```

**Comment by lizziel** [comment]
The tagO3 bug will be fixed in https://github.com/geoschem/geos-chem/pull/1724.

**Review by msulprizio — APPROVED** [review]
This pull request looks good to merge. It replaces several uses of GCARR_abc with GCARR_ac where the B coefficient is zero in the .eqn files.

**Comment by msulprizio** [comment]
Note: This PR causes differences in the full-chemistry benchmark simulation and should be evaluated with it's own alpha tag. We will hold off on merging in until 14.3.0.

**Comment by yantosca** [comment]
I am looking into this.  The only modifications were changing GCARR_abc to GCARR_ac and deleting the B=0.0 term from the equation.  It could be that this is just a difference in optimization.  I will investigate further.

**Comment by yantosca** [comment]
It might be possible that the `K300_OVER_TEMP**b0` term in `GCARR_abc` might have some numerical noise (i.e 0.9999999999 instead of 1.00000000) that might be causing the differences. 
```F90
  FUNCTION GCARR_ac( a0, c0 ) RESULT( k )
    ! Arrhenius function, skipping computation of ( 300/T )**b0,
    ! which evaluates to 1 when b0=0.  This avoids excess CPU
    ! cycles (bmy, 12/18/20)
    !
    REAL(dp), INTENT(IN) :: a0, c0
    REAL(dp)             :: k
    !
    k = a0 * EXP( c0 / TEMP )
  END FUNCTION GCARR_ac

  FUNCTION GCARR_abc( a0, b0, c0 ) RESULT( k )
    ! Arrhenius function, using all 3 terms.
    ! Use this when a0, b0, c0 are all nonzero.
    !
    REAL(dp), INTENT(IN) :: a0, b0, c0
    REAL(dp)             :: k
    !
    k = a0 * EXP( c0 / TEMP ) * K300_OVER_TEMP**b0
  END FUNCTION GCARR_abc
```

**Comment by yantosca** [comment]
After merging up to 14.3.0-alpha.1, all GEOS-Chem Classic integration tests (except TOMAS) passed:
```console
==============================================================================
GEOS-Chem Classic: Execution Test Results

GCClassic #e5e9e48 GEOS-Chem submodule update: Merge PR #2006 containing interim fix for HMS chemistry
GEOS-Chem #a0a6a495a Bring the feature/use-gcarr-ac branch up to 14.3.0-alpha.1
HEMCO     #00aaf65 Update HEMCO 3.7.1 release date in CHANGELOG.md

Using 24 OpenMP threads
Number of execution tests: 27

Submitted as SLURM job: 10370934
==============================================================================

Summary of test results:
------------------------------------------------------------------------------
Execution tests passed: 25
Execution tests failed: 2
Execution tests not yet completed: 0
```

**Comment by yantosca** [comment]
All GCHP integration tests passed
```console
==============================================================================
GCHP: Execution Test Results

GCClassic #2a73088 GEOS-Chem submodule update: Merge PR #2006 containing interim fix for HMS chemistry
GEOS-Chem #a0a6a495a Bring the feature/use-gcarr-ac branch up to 14.3.0-alpha.1
HEMCO     #00aaf65 Update HEMCO 3.7.1 release date in CHANGELOG.md

Number of execution tests: 5

Submitted as SLURM job: 10373279
==============================================================================

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%  All execution tests passed!  %%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
```

File: KPP/fullchem/CHANGELOG_fullchem.md
Comment (msulprizio) [inline KPP/fullchem/CHANGELOG_fullchem.md:44]: Should these lines be removed? I think they're leftover from a merge since they also occur above.

**Review by msulprizio — CHANGES_REQUESTED** [review]
Looks good to merge after cleaning up a few lines in `CHANGELOG_fullchem.md`.

**Comment by yantosca** [comment]
After merging up to 14.3.0, all GEOS-Chem Classic integration tests passed:
```console
==============================================================================
GEOS-Chem Classic: Execution Test Results

GCClassic #347e3de Update version number to 14.3.0 in preparation for release
GEOS-Chem #aebdaade0 Merge PR #1717 (Use GCARR_ac rate-law function where B=0)
HEMCO     #f807e1a Update HEMCO version number to 3.8.0 in preparation for release

Using 24 OpenMP threads
Number of execution tests: 26

Submitted as SLURM job: 19382525
==============================================================================

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%  All execution tests passed!  %%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
```

**Comment by yantosca** [comment]
After merging up to 14.3.0, all GCHP integration tests passed:
```console
==============================================================================
GCHP: Execution Test Results

GCHP      #14cb756 Update version number to 14.3.0 in preparation for release
GEOS-Chem #aebdaade0 Merge PR #1717 (Use GCARR_ac rate-law function where B=0)
HEMCO     #f807e1a Update HEMCO version number to 3.8.0 in preparation for release

Number of execution tests: 6

Submitted as SLURM job: 19430464
==============================================================================
 
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%  All execution tests passed!  %%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
```

**Comment by yantosca** [comment]
Note: This was merged into dev/14.4.0.  I forgot to change the base branch back to dev/14.4.0.

**Pull request closed without merging**
